### DISCUSSION TITLE
Example code

### DISCUSSION THREAD

**Discussion by kanekoshoyu** [eclipse-iceoryx/iceoryx2#67] [Q&A] [answered]
Hi, I saw this project and I am really interested in it; I started trying to set up the pub/sub example, but I ran into this error in the example
```
UnableToCreateDataSegment
```
Also when I tried to just set up the publisher, it says
```
Error: PublishSubscribeOpenError(HangsInCreation)
```

I guess there is a problem with the creation of the server. May I know if there is any documentation that's for the setup that I can look into?

**Comment by kanekoshoyu** [comment]
I did some tracing, seems it's getting stuck at `Config::get_global_config()`
and when I load with `Config::default()` instead, it reports error below
```
        0 [T] FileBuilder { file_path: FilePath { value: FixedSizeByteString<4096> { len: 76, data: "/tmp/iceoryx2/services/iox2_25b25afeb7557886e
              9f69408151e018e268e5917.dynamic" } }, access_mode: ReadWrite, permission: OWNER_READ | OWNER_WRITE | OWNER_EXEC | OWNER_ALL, owner: 
              None, group: None, truncate_size: None, creation_mode: Some(CreateExclusive) }
              | created
        1 [D] SharedMemoryBuilder { name: FileName { value: FixedSizeByteString<255> { len: 53, data: "iox2_25b25afeb7557886e9f69408151e018e268e59
              17.dynamic" } }, size: 422, is_memory_locked: false, has_ownership: false, permission: OWNER_READ | OWNER_WRITE | OWNER_EXEC | OWNER
              _ALL, creation_mode: Some(CreateExclusive), zero_memory: false, access_mode: ReadWrite, enforce_base_address: None }
              | Unable to create shared memory since the name exceeds the maximum supported length of 255.
```

**Reply by kanekoshoyu** [reply]
The filename is just 53 bytes, so I believe there are logical flaws when it reports that it exceeds 255 bytes.

**Reply by elfenpiff** [reply]
@kanekoshoyu On what platform are you on? MacOS or Windows?

I suspect that the platform abstraction layer has maybe an issue. But if you are on Linux, could you please try out the steps described in the faq: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/FAQ.md and cleanup all resources.

How did this error occur? Right from the beginning, or did you run the example once and could not restart it?

**Reply by kanekoshoyu** [reply]
I'm using macOS. Let me try with Linux to see if the problem persists. What is your set up?

There are two patterns, when the .dynamic file does not exist and is first created in the create_or_open() function, it says the file name is too long, which isn't. I suppose the 255 byte filename is meant for actual OS restriction, but I see the actual empty file getting created.

When I run it again, it attempts to open the existing empty .dynamic file and says it's not allowed to open the file while the file is getting created. I suppose there should be some initial data in the first file creation.

**Reply by kanekoshoyu** [reply]
I'm using macOS. Let me try with Linux to see if the problem persists. What is your set up?

There are two patterns, when the .dynamic file does not exist and is first created in the create_or_open() function, it says the file name is too long, which isn't. I suppose the 255 byte filename is meant for actual OS restriction, but I see the actual empty file getting created.

When I run it again, it attempts to open the existing empty .dynamic file and says it's not allowed to open the file while the file is getting created. I suppose there should be some initial data in the first file creation.

**Reply by elfenpiff** [reply, marked as answer]
@kanekoshoyu 

I tried all platforms and could not reproduce the problem, but I may know where the problem lies. The initial 0.1.1 release did not have Mac OS support, this came over Christmas and is currently only supported on the iceoryx2 main branch.

Mac OS has the restriction that a Shared Memory Name must be smaller than 31 characters - this was fixed. Without the fix, you observe exactly the problem you have.

So could you try out if `main` works for you on mac os. If so, I can create a new release 0.1.2 for you :)

**Reply by kanekoshoyu** [reply]
Yeah I just tried using main, I think the initialization is solved! Looking forward to the new release then :)

**Comment by elfenpiff** [comment]
@kanekoshoyu iceoryx2 0.2.1 is out with full mac os support. Have fun.
